builtin/logical/ssh/path_config_ca.go

```diff
@@ -186,127 +193,73 @@ func caKey(ctx context.Context, storage logical.Storage, keyType string) (*keySt
 }
 
 func (b *backend) pathConfigCAUpdate(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
-	var err error
-	publicKey := data.Get("public_key").(string)
-	privateKey := data.Get("private_key").(string)
-
-	var generateSigningKey bool
-
-	generateSigningKeyRaw, ok := data.GetOk("generate_signing_key")
-	switch {
-	// explicitly set true
-	case ok && generateSigningKeyRaw.(bool):
-		if publicKey != "" || privateKey != "" {
-			return logical.ErrorResponse("public_key and private_key must not be set when generate_signing_key is set to true"), nil
-		}
-
-		generateSigningKey = true
-
-	// explicitly set to false, or not set and we have both a public and private key
-	case ok, publicKey != "" && privateKey != "":
-		if publicKey == "" {
-			return logical.ErrorResponse("missing public_key"), nil
-		}
-
-		if privateKey == "" {
-			return logical.ErrorResponse("missing private_key"), nil
-		}
-
-		_, err := ssh.ParsePrivateKey([]byte(privateKey))
-		if err != nil {
-			return logical.ErrorResponse(fmt.Sprintf("Unable to parse private_key as an SSH private key: %v", err)), nil
-		}
-
-		_, err = parsePublicSSHKey(publicKey)
-		if err != nil {
-			return logical.ErrorResponse(fmt.Sprintf("Unable to parse public_key as an SSH public key: %v", err)), nil
-		}
-
-	// not set and no public/private key provided so generate
-	case publicKey == "" && privateKey == "":
-		generateSigningKey = true
+	// Since we're planning on updating issuers here, grab the lock so we've
```

Comment:
Is there some way to unify the code with the other version of this handler a bit more? This goes to the same `b.pathImportIssuers` in PKI (in `pki/path_config_ca.go`). 

`req` gives you `Path` as a field if you need to condition some small behavior difference on a given path. 

Same applies for delete or read.

Reply:
I'm not sure if I like the approach that I have followed but let me know what you think. I introduced a new variable that checks the incoming request path and based on that decides what code to execute.
```go
isConfigCARequest := req.Path == "config/ca"
```